cmd/placement/options/options.go

```diff
@@ -36,6 +37,14 @@ const (
 	defaultPlacementPort     = 50005
 	defaultReplicationFactor = 100
 	envMetadataEnabled       = "DAPR_PLACEMENT_METADATA_ENABLED"
+
+	envKeepAliveTime      = "DAPR_PLACEMENT_KEEPALIVE_TIME"
+	envKeepAliveTimeout   = "DAPR_PLACEMENT_KEEPALIVE_TIMEOUT"
+	envDisseminateTimeout = "DAPR_PLACEMENT_DISSEMINATE_TIMEOUT"
```

Comment:
Why are these environment variables, rather than CLI flags?

Reply:
That was my first approach, too if you look at the commit history, but the initial feedback was that it's better not to expose these values too much because wrong configuration values can substantially degrade the service.

Reply:
We should never silently drop configuration values- this results in users having no idea about whether configuration has actually been applied to the system or not and basically having a miss-trust in the software completely. We have this issue today already in our CRDs like Components and Resiliency.

> wrong configuration values can substantially degrade the service

If the configuration is wrong, we should error exit daprd. We shouldn't try to hide configuration- if the configuration itself has large performance impact, this should be clearly documented as such.

Using CLI flags is the _main_ method for configuring runtime options on daprd. Environment variables should be considered an escape hatch.

Reply:
I agree completely on the failing silently comment, and we actually do that ([example](https://github.com/dapr/dapr/pull/8015/files#diff-ef615d6fc2542a98f506d5b7aae40e875a21770cc30427e8a20f2f942152f453R102)).
When I said "wrong configuration values" I meant values that aren't optimal for one network but would be completely valid for another (a better term would have been "suboptimal configuration values"). Values outside the range of what we consider acceptable will be rejected on init.

I also agree with the documentation part and tried to document it as clearly as possible in this [PR](https://github.com/dapr/docs/pull/4327/files).

Re: CLI flags vs env variables: the feedback was that we want to avoid users seeing the flags exposed and feel enticed to change their values without doing a proper benchmark of their network latency and stability. The counter-argument to that is that users in production systems usually spend a lot of time researching and tweaking every argument, but both of these are just assumptions, of course. I'm open to reverting to flags if the team agrees, that way we keep everything nice and organised.

@artursouza @alicejgibbons  what do you think?

Reply:
The initial reason for using env vars was that this was a very niche option for specific use cases and didn't necessarily need to be "as discoverable" as CLI flags. If we are moving away from env vars to standardize on cli flags then that approach is fine.

There were also concerns that these cli flags might not be backwards compatible, so if they are provided in version 1.15 daprd then downgraded to this could cause errors with older versions.

Reply:
I think the downgrading problem is not specific to these flags; it's a more generic thing we should address on a different level.

Maybe we should just do this the "dapr way" and revert to flags. As @JoshVanL was saying above, having the parameters clearly documented should prevent misconfigurations by users.